po/zh_CN.po

```diff
@@ -172,67 +172,81 @@ msgid ""
 "a logical subset, simply wrap with which(), and take the which() outside the "
 "loop if possible for efficiency."
 msgstr ""
+"i 为 '%s' 类型。i 必须为整（integer）型，或数值（numeric）型（将被强制转换成整型并发出警告）。如果 i 为"
+"一逻辑向量（logical）的子集，请直接将它传给 which()，且如果可能的话将 which() 放置于"
```

Comment:
I think this would mainly come up for something like

```
iris = as.data.table(iris)
set(iris, rep(TRUE, nrow(iris)), 'new_col', rep(5L, nrow(iris)))
```

This should be replaced by

```
set(iris, which(rep(TRUE, nrow(iris))), 'new_col', rep(5L, nrow(iris)))
```

The latter part of the message refers to if `set` is being used in a loop:

```
for (jj in 1:100) set(iris, which(rep(TRUE, nrow(iris))), sprintf("V%d", jj), rnorm(nrow(iris)))
```

Will be more efficient with:

```
idx <- which(rep(TRUE, nrow(iris)))
for (jj in 1:100) set(iris, idx, sprintf("V%d", jj), rnorm(nrow(iris)))
```

Since we don't have to run `which` every iteration.

Reply:
Previously I thought this message came from cases when input of `i` is not the same length of `nrow(dt)` but shorter, and `which(i)` will change it into a row index to make `set` work. So `a logical subet` would mean `a subset of a longer logical vector `.

Now I know the full context. Thanks.

`一个用于筛选的逻辑（logical）向量 ` is definitely a better and more precise one.